content/en/blog/_posts/2023-01-17-confidential-kubernetes.md

```diff
@@ -0,0 +1,324 @@
+or inject arbitrary code and data for the TEE to execute and therefore easily leak critical
+information. **Integrity** guarantees a TEE owner that neither code nor data can be
+tampered with while running critical computations.
+
+*Availability* is the third basic property often discussed in the context of information
+security. This property is outside the scope of most TEEs. Usually, they can be controlled
+(shut down, restarted, …) by some higher level abstraction. This could be the CPU itself, the
+hypervisor, or the kernel. This is to preserve the availability of the overall system, and
+not the TEE itself. When running in the cloud availability is usually guaranteed by
+the cloud provider in terms of Service Level Agreements (SLAs) and is not cryptographically enforceable.
+
+Confidentiality and Integrity by themselves would only be helpful in some cases. For example,
+consider a TEE running in a remote cloud. How would you know that the TEE is genuine and running
+your intended software? It could be an imposter stealing your data as soon as you send it over.
+This fundamental problem is addressed by **Attestability**. Attestation allows us to verify
+the identity, confidentiality, and integrity of TEEs based on cryptographic certificates issued
+from the hardware itself. This feature can also be made available to clients outside of the
+confidential computing hardware in the form of remote attestation.
+
+TEEs can hold and process information that predates or outlives the trusted environment. That
+could mean across restarts, different versions, or platform migrations. Therefore **Recoverability**
+is an important feature. Data and the state of a TEE need to be sealed before they are written
+to persistent storage to maintain confidentiality and integrity guarantees. The access to such
+sealed data needs to be well-defined. In most cases, the unsealing is bound to a TEE's identity.
+Hence, making sure the recovery can only happen in the same confidential context.
+
+This does not have to limit the flexibility of the overall system. The
+[migration agent (MA) of AMD SEV-SNP](https://www.amd.com/system/files/TechDocs/SEV-SNP-strengthening-vm-isolation-with-integrity-protection-and-more.pdf)
+allows users to migrate a confidential virtual machine to a different host system
+while keeping the security properties of the TEE intact.
+
+## Feature comparison
+
+In the following, we will dive a little bit deeper into the specific implementations,
+compare supported features and analyze their security properties.
+
+### AMD SEV
+
+AMD's [Secure Encrypted Virtualization (SEV)](https://developer.amd.com/sev/) technologies
+are a set of features to enhance the security of virtual machines on AMD's server CPUs. SEV
+transparently encrypts the memory of each VM with a unique key. SEV can also calculate a
+signature of the memory contents, which can be sent to the VM's owner as an attestation that
+the memory was encrypted correctly by the firmware.
```

Comment:
The attestation is about more than ensuring that the firmware correctly encrypted the initial guest memory. I think the main idea is that it guarantees to the guest owner that the initial guest memory is what the expect.